pkg/sql/logictest/testdata/logic_test/views

```diff
@@ -380,68 +380,170 @@ CREATE VIEW virt2 AS SELECT range_id, lease_holder FROM crdb_internal.ranges
+statement ok
+CREATE VIEW star11 AS SELECT t1.a AS a1, t2.* FROM t AS t1 JOIN t AS t2 ON t1.a = t2.a
 
-statement error views do not currently support \* expressions
-create view s1 AS SELECT t1.a, t2.a FROM t AS t1 JOIN (SELECT * FROM t) AS t2 ON t1.a = t2.a
+# This error is the same as in postgres.
+statement error duplicate column name: "a"
+CREATE VIEW star12 AS SELECT t1.*, t2.* FROM t AS t1 JOIN t AS t2 ON t1.a = t2.a
 
-statement error views do not currently support \* expressions
-create view s1 AS SELECT t1.a, t2.a FROM t AS t1 JOIN t AS t2 ON t1.a IN (SELECT a FROM (SELECT * FROM t))
+# This error is the same as in postgres.
+statement error duplicate column name: "a"
+CREATE VIEW star13 AS SELECT * FROM t AS t1 JOIN t AS t2 ON t1.a = t2.a
 
 statement ok
-create view s1 AS SELECT count(*) FROM t
+CREATE VIEW star14 AS SELECT t1.a, t2.a AS a2 FROM (SELECT * FROM t) AS t1 JOIN t AS t2 ON t1.a = t2.a
 
 statement ok
-create view s2 AS SELECT a FROM t WHERE a IN (SELECT count(*) FROM t)
+CREATE VIEW star15 AS SELECT t1.a, t2.a AS a2 FROM t AS t1 JOIN (SELECT * FROM t) AS t2 ON t1.a = t2.a
 
 statement ok
-create view s3 AS SELECT a, count(*) FROM t GROUP BY a
+CREATE VIEW star16 AS SELECT t1.a, t2.a AS a2 FROM t AS t1 JOIN t AS t2 ON t1.a IN (SELECT a FROM (SELECT * FROM t))
 
-statement ok
-create view s4 AS SELECT a, count(*) FROM t GROUP BY a HAVING a > (SELECT count(*) FROM t)
+query TT
+SELECT descriptor_name, create_statement FROM crdb_internal.create_statements WHERE descriptor_name LIKE 'star%' ORDER BY 1
+----
+star1   CREATE VIEW public.star1 (
+          a,
+          b
+        ) AS SELECT t.a, t.b FROM test.public.t
+star10  CREATE VIEW public.star10 (
+          a,
+          b,
+          a2
+        ) AS SELECT t1.a, t1.b, t2.a AS a2 FROM test.public.t AS t1 JOIN test.public.t AS t2 ON t1.a = t2.a
+star11  CREATE VIEW public.star11 (
+          a1,
+          a,
+          b
+        ) AS SELECT t1.a AS a1, t2.a, t2.b FROM test.public.t AS t1 JOIN test.public.t AS t2 ON t1.a = t2.a
+star14  CREATE VIEW public.star14 (
+          a,
+          a2
+        ) AS SELECT
+            t1.a, t2.a AS a2
+          FROM
+            (SELECT t.a, t.b FROM test.public.t) AS t1 JOIN test.public.t AS t2 ON t1.a = t2.a
+star15  CREATE VIEW public.star15 (
+          a,
+          a2
+        ) AS SELECT
+            t1.a, t2.a AS a2
+          FROM
+            test.public.t AS t1 JOIN (SELECT t.a, t.b FROM test.public.t) AS t2 ON t1.a = t2.a
+star16  CREATE VIEW public.star16 (
+          a,
+          a2
+        ) AS SELECT
+            t1.a, t2.a AS a2
+          FROM
+            test.public.t AS t1
+            JOIN test.public.t AS t2 ON t1.a IN (SELECT a FROM (SELECT t.a, t.b FROM test.public.t))
+star2   CREATE VIEW public.star2 (
+          a,
+          b
+        ) AS SELECT t.a, t.b FROM test.public.t
+star3   CREATE VIEW public.star3 (
+          a
+        ) AS SELECT a FROM test.public.t ORDER BY t.*
```

Comment:
I am noting how the stars inside ORDER BY and GROUP BY are preserved here. This means they are also perserved in UDFs.

I believe this is OK? But I think we need some semantic expert to double check. In any case, this choice is not specific to this PR so it should not block it.